common/menu.c

```diff
@@ -73,6 +107,19 @@ void menuHandleAButton(void) {
     }
 }
 
+void menuHandleXButton(void) {
+    menu_s* menu = menuGetCurrent();
+    
+    if (menu->nEntries > 0 && hbmenu_state == HBMENU_DEFAULT)
+    {
+        int i;
+        menuEntry_s* me;
+        for (i = 0, me = menu->firstEntry; i != menu->curEntry; i ++, me = me->next);
```

Comment:
`for (int i = 0, etc etc)`

Reply:
I copy/pasted this verbatim from the function above it, menuHandleAButton - shall I change it there too?

Reply:
Actually this change breaks it, because of the additional assignment after the comma.

> common/menu.c:112:30: warning: initialization of 'int' from 'menuEntry_s *' {aka 'struct menuEntry_s_tag *'} makes integer from pointer without a cast [-Wint-conversion]
         for (int i = 0, me = menu->firstEntry; i != menu->curEntry; i ++, me = me->next);
                              ^~~~
common/menu.c:112:82: error: invalid type argument of '->' (have 'int')
         for (int i = 0, me = menu->firstEntry; i != menu->curEntry; i ++, me = me->next);

Reply:
So I left them both unchanged, is that ok?

Reply:
Actually yeah nevermind, I didn't see that there's another variable (of different type) that is being assigned to.